Update documentation (#17)
lendi-au/StackJanitor: README.md

# StackJanitor

> StackJanitor is a tool to clean up AWS CloudFormation Stacks in development environment based on TTL (expiration time) and git pull request events. It cleans up unused development CFN stacks to help you save AWS cloud development costs.

Reviewer: ```suggestion
![StackJanitor](https://i.imgur.com/KMkUEo6.jpg)
StackJanitor is a tool to clean up AWS CloudFormation Stacks in development environment based on TTL (expiration time) and git pull request events. It cleans up unused development CFN stacks to help you save AWS cloud development costs.
```

# StackJanitor

![StackJanitor](https://i.imgur.com/KMkUEo6.jpg)

StackJanitor is a tool to clean up AWS CloudFormation Stacks in development environment based on TTL (expiration time) and git pull request events. It cleans up unused development CFN stacks to help you save AWS cloud development costs.

## How it works

StackJanitor automatically monitors all the CloudFormation stacks that you create in the AWS account. If stackjanitor tag is enabled, StackJanitor will put a default TTL (based on your configurations) for the Stack. It will clean up the CFN Stack when TTL is expired.

**Example:** You have deployed a development CFN Stack with a tag `stackjanitor = "enabled"`. if the default TTL is set as 7 days (604800 seconds) in the `serverless.yml`, the CFN stack will be cleaned up automatically after 7 days. Any `UpdateStack` will refresh the TTL.

![StackJanitor Architecture](./StackJanitor.png "StackJanitor Architecture")

> StackJanitor also works with Github and Bitbucket Pull request events. If your CloudFormation Stack has tag for specific git branch and repo, StackJanitor can clean up the Stack when pull request is merged or declined for the branch.

Reviewer: ```suggestion
StackJanitor can also delete stacks using either a Github or Bitbucket webhook. Our current workflow uses merged pull request events to trigger the webhook so that we can quickly delete stacks as part of our development workflow.

To use this feature, your CloudFormation Stack must also be tagged with `BRANCH` and `REPOSITORY` in order for StackJanitor to find the stack associated with your branch/repository combination and delete it.
```

Set up default expiration period (TTL) in `serverless.yml` custom vars.

## Setting up the Git Webhook

Tag your CloudFormation Stack,
REPOSITORY = `Git repo name` and
BRANCH = `Git branch`.

> When you deploy the StackJanitor by running `sls deploy` command, you should receive a webhook endpoint output url in your terminal.

Reviewer: ```suggestion
When you deploy StackJanitor by running the `sls deploy` command, you should receive a webhook endpoint output url in your terminal.
```

Now, browse the webhook settings in your git repository and add the webhook enpoint.

> If you are using **BitBucket,**

Reviewer: ```suggestion
### Bitbucket
```

1. Go to `Settings > Webhooks` and click `Add Webhook`.
2. Put the webhook endpoint in `URL` field.
3. Set `Status` to `Active`.
4. Set `Triggers` to `Choose from a full list of triggers` and select Pull request `Merged` and `Declined`.
5. Save the settings.

> If you are using **GitHub,**

Reviewer: ```suggestion
### GitHub
```
